hack/make/validate-vendor

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+source "${MAKEDIR}/.validate"
+
+IFS=$'\n'
+files=( $(validate_diff --diff-filter=ACMR --name-only -- 'hack/vendor.sh' 'vendor/' || true) ) 
+unset IFS
+
+if [ ${#files[@]} -gt 0 ]; then
+	# We run vendor.sh to and see if we have a diff afterwards
+	./hack/vendor.sh >/dev/null
+	# Let see if the working directory is clean
+	diffs="$(git status --porcelain -- vendor 2>/dev/null)"
+	if [ "$diffs" ]; then
+		{
+			echo 'The result of ./hack/vendor.sh differs'
+			echo
+			echo $diffs
```

Comment:
you may want to tweak the formatting of this so that each file appears on its own line.
right now it shows as:
```
23:01:35 ---> Making bundle: validate-vendor (in bundles/1.10.0-dev/validate-vendor)

23:04:11 The result of ./hack/vendor.sh differs
23:04:11 
23:04:11 M vendor/src/github.com/mistifyio/go-zfs/CONTRIBUTING.md M vendor/src/github.com/mistifyio/go-zfs/utils.go M vendor/src/github.com/mistifyio/go-zfs/zfs.go M vendor/src/github.com/mistifyio/go-zfs/zpool.go
23:04:11 
23:04:11 Please vendor your package with ./hack/vendor.sh.

```